### PR TITLE
Add NUMA build to CI

### PR DESCRIPTION
**Pull request by davidbeckingsale**

### CODE REVIEW

**Comment by davidbeckingsale** [comment]
LGTM

**Comment by davidbeckingsale** [comment]
LGTM

**Comment by davidbeckingsale** [comment]
LGTM

**Review by kab163 — APPROVED** [review]
Pretty straight forward. Looks good

File: src/umpire/ResourceManager.cpp [hunk 1]
```diff
@@ -20,6 +20,9 @@
 #include "umpire/strategy/DynamicPool.hpp"
 #include "umpire/strategy/FixedPool.hpp"
 #include "umpire/strategy/ZeroByteHandler.hpp"
+#if defined(UMPIRE_ENABLE_NUMA)
+#include "umpire/strategy/NumaPolicy.hpp"
+#endif
```
Comment (mcfadden8) [inline src/umpire/ResourceManager.cpp:25]: It seems strange (to me) that the only change made to this file was to add a guarded include of NumaPolicy.hpp.  Is there someplace else further down in this file that should also be guarded?

Comment (davidbeckingsale) [reply src/umpire/ResourceManager.cpp:25]: I think this got accidentally removed at some point. The `move` operation uses this class.

Comment (mcfadden8) [reply src/umpire/ResourceManager.cpp:25]: Ah.  I guess we wern't building with NUMA very much.  Glad it has been added to CI!

**Review by mcfadden8 — APPROVED** [review]

**Pull request merged**